fix: Correct iOS Contents.json for SVG and misc updates

peteranny opened the pull request:
This PR addresses issues with iOS asset generation:

- Add `cleanAssetDirectory` function to prevent asset conflicts
- Refactor `processImageForScale` to use named parameters
- Revise `createContentsJson` to support different formats (png/svg)
- Add tracking of processed components to identify unprocessed ones
- Change warning for missing components from yellow to red text

This improves iOS asset export workflow and provides better error visibility.

# Testing Steps

```sh
FIGMA_TOKEN="67747-b4795e9e-26fd-4f69-b538-ac7dec5babac" node src/index.js icon/verified button/arrow_blue img/illustration_afterlogin img/ani_wiggle
```

peteranny committed:
image uses format

peteranny committed:
readability

peteranny committed:
universal svg

peteranny committed:
clean before creation

peteranny committed:
report unproccessed

peteranny committed:
buttons

peteranny committed:
Merge remote-tracking branch 'upstream/main' into fix/ios-scaled-assets

peteranny committed:
simplify template

peteranny committed:
remove unneeded comments

peteranny committed:
Revert "buttons"

This reverts commit a1aba7eee078c62620dbc214af5f0a447dca14d1.

peteranny committed:
suffix -id

peteranny commented:
@hungyanbin Can you help merge this branch? I don't have permission. 🙏🏼

yyjim commented on src/index.js:
```diff
@@ -358,7 +358,7 @@ async function fetchComponents(fileId, componentNames, pageId = '', pageName = '
       const notFound = componentNames.filter(name => !nameMap.has(name));
       if (notFound.length > 0) {
-        console.log(chalk.yellow(`Warning: The following components were not found: ${notFound.join(', ')}`));
+        console.log(chalk.red(`Warning: The following components were not found: ${notFound.join(', ')}`));
```
If we change this to red, should it be treated as an error?

peteranny replied:
No, it just stands out.

![Screenshot 2025-05-15 at 3 54 10 PM](https://github.com/user-attachments/assets/0c91195c-5945-4979-9f0d-759144a0ba17)

yyjim replied:
Warning: to me implies a yellow color, not red.

This is minor—I don’t have a strong opinion about making this change.

yyjim commented on src/index.js:
```diff
@@ -649,28 +668,49 @@ async function processImageForScale(imageBuffer, scale, format, quality, maxScal
  * Create Contents.json for iOS assets
  */
-function createContentsJson(name, type = 'image') {
+function createContentsJson(name, format = 'png') {
```
suggestion: Should we consider making the function name more related to iOS?

eg. `createAssetCatalogContentsJson`

peteranny replied:
Sounds good, but I'd like to keep this PR smaller from unrelated changes. Let's find a chance in some future PR to update it.

yyjim commented on src/index.js:
```diff
@@ -649,28 +668,49 @@ async function processImageForScale(imageBuffer, scale, format, quality, maxScal
         idiom: 'universal',
         scale: '1x'
       },
       {
-        filename: `${name}@2x.png`,
+        filename: `${name}@2x.${format}`,
         idiom: 'universal',
         scale: '2x'
       },
       {
-        filename: `${name}@3x.png`,
+        filename: `${name}@3x.${format}`,
         idiom: 'universal',
         scale: '3x'
       }
     ],
     info: {
-      author: 'Figma Asset Downloader',
-      version: 1
+      version: 1,
```
Right now it’s missing handling for the ~ipad case.

peteranny replied:
I thought universal covers either device?

yyjim replied:
This is an example

```
{
  "info" : {
    "author" : "xcode",
    "version" : 1
  },
  "images" : [
    {
      "filename" : "button_context_curve~ipadL@1x.png",
      "idiom" : "ipad",
      "scale" : "1x"
    },
    {
      "scale" : "2x",
      "filename" : "button_context_curve~ipadL@2x.png",
      "idiom" : "ipad"
    },
    {
      "scale" : "1x",
      "filename" : "button_context_curveL@1x.png",
      "idiom" : "universal"
    },
    {
      "filename" : "button_context_curveL@2x.png",
      "scale" : "2x",
      "idiom" : "universal"
    },
    {
      "scale" : "3x",
      "idiom" : "universal",
      "filename" : "button_context_curveL@3x.png"
    }
  ]
}
```

peteranny replied:
Current script only retrieves a baseline image and does scale itself, ie. `function processImageForScale`, so the iPad version does no difference yet. I'll open another PR for this fix.

peteranny replied:
Opened #5 to fix this

landicefu reviewed (changes requested):
I did some test and find some problems.
1. node src/index.js icon/e_align_center
This doesn't include image assets, and it's throwing error
```
An error occurred: processedImageIds is not iterable
```

2. We are using webp on Android, so the format in the config will be webp. However, I get this now
```
✖ Error getting image URLs from Figma
Status: 400
Message: {"status":400,"err":"Parameter 'format' must be one of 'svg', 'png', or 'jpg', 'pdf'"}
```

3. If I request for 3 icons now. It will say processing 1, 2, 3, but in the folder there is only one. Is this a bug of the clear folder process. Also, for Android, we don't expect clearing folder. We expect 3 new or updated asset to be merged with the current assets.

peteranny committed:
ipad

peteranny committed:
fix return values

hungyanbin commented on src/index.js:
```diff
@@ -735,25 +777,26 @@ async function processIcons(components, fileId, config) {
           const xmlContent = await convertSvgForPlatform(optimizedSvg, 'android');
           const drawablePath = path.join(config.icons.path, 'drawable');
-          await fs.ensureDir(drawablePath);
+          await cleanAssetDirectory(drawablePath);
```
Suggestion: We will use existing figma asset folder, this will cause all of our asset being cleared, to prevent conflict, maybe ask user that he want to override or not?

landicefu replied:
Or maybe add a config to explicitly decide if we clear the target folder or not

peteranny replied:
Updated - didn't know Android put all assets of the same dpi in the same place. Updated to not clear the directory.

peteranny committed:
revert png download

peteranny committed:
ids -> names

peteranny committed:
clean asset dirs only for ios

peteranny committed:
report unfetched input names

peteranny commented:
> 1. node src/index.js icon/e_align_center
>    This doesn't include image assets, and it's throwing error

Fixed the return values. Thanks for catching!

>  2. We are using webp on Android, so the format in the config will be webp. However, I get this now

Fixed, it's supposed to retrieve png. Thanks for catching!


>  3. If I request for 3 icons now. It will say processing 1, 2, 3, but in the folder there is only one. Is this a bug of the clear folder process. Also, for Android, we don't expect clearing folder. We expect 3 new or updated asset to be merged with the current assets.

Fixed. It now clears the directory only for iOS.
